### ISSUE TITLE
Screen reader focus order is not in sequential order under SpinButton with labelPosition=Position.end

### ISSUE THREAD

**Issue by CdricGmd** [OfficeDev/office-ui-fabric-react#11728]
### Environment Information

- **Package version(s)**: 7.82.1
- **Browser and OS versions**: Windows 10 1909, Microsoft Edge 44.18362.449.0 (Spartan), Microsoft Edge 81.0.394.0 (Chromium)

### Describe the issue:

Screen reader focus order is not in sequential order under SpinButton with labelPosition=Position.end.

### Please provide a reproduction of the issue in a codepen:

Add a `SpinButton` component with prop `labelPostion={Position.end}`, and some text before and after the component.

[https://codepen.io/mstest_cdg/pen/wvBEoxJ](https://codepen.io/mstest_cdg/pen/wvBEoxJ)

![SpinButton with Label at Position End](https://user-images.githubusercontent.com/3540886/72508346-cf4c3b80-3845-11ea-91fb-1e47d40ba25a.gif)

Also reproduces on "Basic SpinButton With Icon and Positioned at the End" sample of  [https://fabricweb.z5.web.core.windows.net/oufr/7.82.1/#/examples/spinbutton](https://fabricweb.z5.web.core.windows.net/oufr/7.82.1/#/examples/spinbutton)

#### Actual behavior:

Screen reader focus order is the following
1. "Text before"
2. SpinButton label
3. SpinButton value input
4. "Text after"

#### Expected behavior:

Screen reader focus order should be the following
1. "Text before"
2. SpinButton value input
3. SpinButton label - _because at the 'end' position._
4. "Text after"

**Comment by xugao** [comment]
Screen reader is scanning in DOM element order. currently the positioning is down purely using css.

**Comment by xugao** [comment]
@CdricGmd - i thought about this more. Could you tell me why you think this affects the user negatively? The label is used to describe the spinbutton. I think in case of using the screen reader, it is the correct experience for user to always hear the label first, despite how it's visually presented.

**Comment by CdricGmd** [comment]
@xugao my concern is specific to _scan mode_, Narrator would read elements in the DOM order. Otherwise I agree screen readers can chose in which order they read labelled elements as long as they're properly annotated.
With `Postion.end`, the label is still before the input in actual DOM. As a counter example: SpinButton with `Position.botton`, in this case, the label appears _after_ the input in the dom, thus the screen readers can follow the right order. Reference: [SpinButton.tsx#L253-L262](https://github.com/OfficeDev/office-ui-fabric-react/blob/7927ef8fc65813b349d97f30fd81be76babd03ef/packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx#L253-L262)

I've worked on a fix to have a proper DOM ordering, respecting the visual ordering, but I'm afraid I don't know CSS enough to change layout to get rid of float and achieve visual parity [https://github.com/CdricGmd/office-ui-fabric-react/commit/a393fa552c869455e4b7871343bd0aedf5e35758](https://github.com/CdricGmd/office-ui-fabric-react/commit/a393fa552c869455e4b7871343bd0aedf5e35758).

**Comment by xugao** [comment]
@CdricGmd - I don't quite follow. Yes - when using narrator (scan mode), narrator read elements in DOM order. My point was - the current behavior is correct for blind users since the visual presentation (how elements are ordered visually, different from actual DOM order) here doesn't matter. the **actual behavior** you described above is a correct experience for narrator users.

**Comment by cnamal** [comment]
Hey folks! 👋
I work with @CdricGmd, here's my 2 cents on the issue: accessibility isn't just about blind people; visually impaired people also use screen readers and having the visual layout close to the DOM one is therefore important.

That being said, we ended dropping our usage of this component because we had a tight SLA to fix our bug. Feel free to close :)

**Comment by CdricGmd** [comment]
@xugao I agree with cnamal, I think fixing this would benefit all accessibility tools users.
Again feel free to close as we worked around the issue on our side.

**Comment by xugao** [comment]
@cnamal , @CdricGmd  - Sorry to hear you guys have to drop the usage to avoid this issue :(  Changing the DOM structure can potentially break existing users. That's why I wanted to make sure this fix is absolutely necessary. do you happen to have some documentations further explain why this fix is needed for visually impaired people? thank you!
